plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/DeploymentResourceMapper.java

```diff
@@ -129,24 +124,12 @@ private void load(String kind) {
 
 	private void buildDeployments() {
 		if (state.compareAndSet(State.UNINITIALIZED, State.LOADING)) {
-			for (String kind : WatchManager.KINDS) {
-				load(kind);
-			}
 			WatchManager.getInstance().stopWatch(project);
+			ConnectionsRegistrySingleton.getInstance().removeListener(connectionListener);
```

Comment:
While mapper builds deployments it does not listen to changes in the project. But while it iterates over kinds of resources, some changes to already loaded kinds can happen in the project, even if with a small probability. Could we add for the time of loading another listener that would just register the fact of a change. If a change is detected, then loading would be repeated.